champ/node/validation/block.rs

```diff
@@ -99,36 +100,41 @@ async fn verify_transactions(
     state: &ChampStateArc,
 ) -> Result<(), BlockValidationError> {
     debug!("verify transactions");
-    let db = &state.db.lock().await;
     // go through all tx in the block and do math to see new balance
     // check against block balance
-    let new_data = new_block.data.as_ref().ok_or(Node::BlockDataNotFound)?;
-    let prev_data = prev_block.data.as_ref().ok_or(Node::BlockDataNotFound)?;
+    let new_data = new_block.data.clone().ok_or(Node::BlockDataNotFound)?;
+    let prev_data = prev_block.data.clone().ok_or(Node::BlockDataNotFound)?;
 
     let mut transaction_ids: Vec<[u8; 32]> = vec![];
 
     if new_data.transactions.len() > 255 {
         return Err(Validation::TooManyTransactions.into());
     }
 
-    // TODO: Run concurrently
     let mut new_balance: i128 = prev_data.balance as i128;
-    for transaction in &new_data.transactions {
+    let mut tokio_tasks: Vec<JoinHandle<Result<i128, BlockValidationError>>> = vec![];
+
+    for t_action in &new_data.transactions {
```

Comment:
maybe change this to tx?

Reply:
I use tx later so transactions makes more sense.